src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    /// <summary>
+    ///     Verifies that the project was loaded with the expected capabilities to catch
+    ///     when the project type in the solution does not match the project itself, for
+    ///     example when the user has renamed csproj -> vbproj without updating the project 
+    ///     type in the solution.
+    /// </summary>
+    [Export(typeof(IVetoProjectPreLoad))]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal partial class ProjectCapabilitiesMissingVetoProjectLoad : IVetoProjectPreLoad
+    {
+        private static readonly ImmutableArray<ProjectType> s_projectTypes = GetProjectTypes();
+        private readonly UnconfiguredProject _project;
+        private readonly IProjectCapabilitiesService _projectCapabilitiesService;
+
+        [ImportingConstructor]
+        public ProjectCapabilitiesMissingVetoProjectLoad(UnconfiguredProject project, IProjectCapabilitiesService projectCapabilitiesService)
+        {
+            _project = project;
+            _projectCapabilitiesService = projectCapabilitiesService;
+        }
+
+        public Task<bool> AllowProjectLoadAsync(bool isNewProject, ProjectConfiguration activeConfiguration, CancellationToken cancellationToken = default)
```

Comment:
This will never return false.

Reply:
Yes.

Reply:
Never mind. I see the interface's docs.

Reply:
I didn't realise this implemented an interface. I was hinting at changing the return type.